lib/msal-common/src/client/AuthorizationCodeClient.ts

```diff
@@ -58,20 +65,36 @@ export class AuthorizationCodeClient extends BaseClient {
             this.logger
         );
 
+        // Validate response. This function throws a server error if an error is returned by the server.
         responseHandler.validateTokenResponse(response.body);
-        const tokenResponse = responseHandler.generateAuthenticationResult(response.body, this.defaultAuthority);
+        const tokenResponse = responseHandler.generateAuthenticationResult(response.body, this.authority, cachedNonce, cachedState);
 
         return tokenResponse;
     }
 
+    /**
+     * Handles the hash fragment response from public client code request. Returns a code response used by
+     * the client to exchange for a token in acquireToken.
+     * @param hashFragment
+     */
+    public handleFragmentResponse(hashFragment: string, cachedState: string): string {
```

Comment:
This is only applicable to SPAs, so it might make sense to have a SPAClient that wraps AuthCode Client, and adds handleFragmentResponse and EndSession API.

Reply:
This is a convenience function that can be extended to check for query params as well, so I think it is ok to keep it here.